src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/DependencyInjectionEventSourceTests.cs

```diff
@@ -343,33 +331,77 @@ public void EmitsServiceProviderBuilt()
         [Fact]
         public void EmitsServiceProviderBuiltOnAttach()
         {
-            _listener.DisableEvents(DependencyInjectionEventSource.Log);
-
             ServiceCollection serviceCollection = new();
             serviceCollection.AddSingleton(new FakeDisposeCallback());
 
             using ServiceProvider provider = serviceCollection.BuildServiceProvider();
 
-            Assert.Empty(_listener.EventData);
+            using var listener = new TestEventListener();
+            listener.EnableEvents(DependencyInjectionEventSource.Log, EventLevel.Verbose);
+            try
+            {
+                EventWrittenEventArgs serviceProviderBuiltEvent = listener.EventDataFor(provider).Single(e => e.EventName == "ServiceProviderBuilt");
+                Assert.Equal(1, GetProperty<int>(serviceProviderBuiltEvent, "singletonServices"));
 
-            _listener.EnableEvents(DependencyInjectionEventSource.Log, EventLevel.Verbose);
+                EventWrittenEventArgs serviceProviderDescriptorsEvent = listener.EventDataFor(provider).Single(e => e.EventName == "ServiceProviderDescriptors");
+                Assert.NotNull(JObject.Parse(GetProperty<string>(serviceProviderDescriptorsEvent, "descriptors")));
```

Comment:
Is this the only use of Newtonsoft? Can we use STJ instead?

Reply:
This line used to be on line 361. I did not change it. Git was just unable to track it. I would prefer not having this unrelated change in this PR.